contributing/giving-constructive-criticism.rst

```diff
@@ -0,0 +1,155 @@
+Giving constructive criticism
+=============================
+
+:doc:`Reviewing issues and pull requests </contributing/community/reviews>`
+is a great way to get started with contributing to the Symfony community.
+Anyone can do it! But before you give a comment, take a step back and think,
+is what you are about to say actually what you intend?
+
+Communicating over the Internet with nothing but text can pose a
+big challenge, especially if you remember that the Symfony community
+is world-wide and is composed of a wide variety of people with differing
+ideas and opinions.
+
+Not everyone speaks English or is able to use a keyboard, some might
+have Dyslexia or similar conditions that affect their writing.
+
+Not to mention that some might have a bad experience from previous
+contributions (to other projects).
+
+You're not alone in this. This guide will try to help you write
+constructive, respectful and helpful reviews and replies.
+
+.. tip::
+
+    This guide is not about lecturing you to "conform" or give-up
+    your ideas and opinions but helping you to better communicate,
+    prevent possible confusion, and keeping the Symfony community a
+    welcoming place for everyone. **You are free to disagree with
+    someone's opinions, just don't be disrespectful.**
+
+First of, accept that many programming decisions are opinions.
+Discuss trade offs, which you prefer, and reach a resolution quickly.
+It's not about being right or wrong, but using what works.
+
+Tone of voice
+-------------
+
+We don't expect you to be completely formal, or to even write error-free
+English. Just remember one thing: Don't swear, and be respectful to others.
+
+Don't reply in anger or with an aggressive tone. You're angry, we understand
+that, but swearing/cursing and name calling doesn't really encourage anyone to
+help you. Take a deep breath, count to 10 and try to *clearly* explain what problems
+you encounter.
+
+Gender pronouns
+---------------
+
+While not "formally" required it's better to use gender-neutral pronouns.
+Unless someone "indicated" their pronouns, use "they", "them" instead of
+"he", "she", "his", "hers", "his/hers", "he/she", etc.
+
+And instead of saying "guys" it's better to use "folks", or Symfonions
```

Comment:
It's the first time I read the word *"Symfonions"*. I'd suggest to remove it because nobody is really using it.

Reply:
looks like a typo of `Symfonians` which is used

Reply:
The "guys" part is in my opinion a hypercorrection, which is something you should explicitly strive to avoid. The word is [dictionary-defined to be a "a colloquial form of addressing a group of people"](https://en.wiktionary.org/wiki/guys), and in that form explicitly gender-neutral.

[Merriam Webster goes even further in explicitly confirmed the gender-neutrality](https://www.merriam-webster.com/dictionary/guys):

> used in plural to refer to the members of a group regardless of sex. *example: I saw her and the rest of the guys*

The great danger of being politically over-correct in areas like this is that people like me, who are irritated by the invalid classification of a normal word as being sexist, then usually have the psychologically very normal [cognitive bias](https://en.wikipedia.org/wiki/List_of_cognitive_biases) of discarding all previous paragraphs they just read.

Reply:
> The "guys" part is in my opinion a hypercorrection

Yeah, but you and I are not a part of the group of people that feel excluded by its use … we're also both from _low context cultures_ and both unimpressed by _politically over-correctness_ :wink: 

I go back to my comment in my review, what we need to clearly state here is this is not about people feeling offended as that is highly subjective. Rather it is about people not feeling **excluded**, and when you "translate" this into how it might be received by a person from a high-context culture it is pretty easy to invoke a feeling of exclusion.

Reply:
We can debate all that we want about this, but dictionary definitions are clear and @curry684's comments are correct: "guy" in singular is not inclusive and refers just to 1 man; "guys" in plural is inclusive and means "a group of people" not "a group of men". And by the way, it's common to use "folks" to refer to someone's parents.

In any case, diversity people are pushing hard to ban "guys" and replace it with "folks" ... so we can't make them change their mind. At least we can say that *"we recommend to use folks"*, instead of saying that *"it's better to use folks"*.

Reply:
I'd be fine with a rewording to the effect of: *Try to avoid using wording that may be considered excluding by non-native speakers, like "hey guys", which is in itself gender-neutral, but may be interpreted as exclusively male by people from other cultures or with other native languages. Consider using less ambiguous alternatives like "Hey folks" or "Thanks peeps".*

This way you convey both understanding of the possible misunderstandings, while neither insulting those who do nor offending those who see no harm, and take no moral stance on accidental use.

Reply:
I think that would adequately address concerns some people might have, @curry684 :+1:

Reply:
This discussion is showing as outdated now since the new commit but not concluded yet I think  :)

Reply:
Briefly chiming in on the topic, since I also brought this up in the initial announcement.

Quoting from https://www.vox.com/2015/6/11/8761227/you-guys-sexism-language
```
"Because male-based generics are another indicator — and more importantly, a reinforcer — of a system in which "man" in the abstract and men in the flesh are privileged over women. Some say that language merely reflects reality and so we should ignore our words and work on changing the unequal gender arrangements that are reflected in our language. Well, yes, in part.
```

Similarly there was a move from "firemen" to "firefighter" etc.

I should also note that language is not fixed, it evolves. So the point of moving away from "guys" for mixed gender (or even women only) groups is that its based on the notion that men are the default. So as we are trying to get away from this notion, it also makes sense that we adjust our language accordingly.

Now, this is something that I mostly see happening among native speakers, ie. people that have mastered the language, and as such have the luxury to think in terms of deliberately evolving the language. Non-native speakers rather are more often surprised when they are told (unfortunately sometimes also in needlessly harsh terms) that they should not use "guys". For non-native speakers struggling to speak fluently its hard to then expect of them to also follow such evolution of language. As such it would be up to the native speakers to simply set an example, which will naturally have non-native speakers learn from this example.

As such I also using @javiereguiluz formulation of `we recommend to use folks` because I think its less judgmental terminology.

Reply:
@lsmith77 fair enough and good points, but our goal as "the Symfony community" should not be to be frontline fighters for the demasculinization of all languages worldwide. As stated before the goal is to prevent perceived exclusion. I think a general recommendation bringing minor attention to the underlying issue is the strongest statement we should want to make on something like this.

Reply:
I actually do think we need to be on the frontline there, because we want to be an example for the rest of the open source community and even the world. However we shouldn't be dogmatic of it in the sense we expect people to not use guys for groups not just consisting of men but we should recommend people to think about this and try to go beyond the status quo, because the status quo is not diversity friendly.

Again I think `recommend` strikes the appropriate balance here to prevent ostracizing people  who lack the english fluency to easily switch to other alternatives and also acknowledging that even people with sufficient fluency it will take time to adapt.

That being said, this topic should not be the one stopping this PR and maybe we can also move this topic to another PR that follows this PR.

Reply:
Like I said I agree with the underlying points, but it's mostly a matter of grammar now. I see my suggestion has been taken verbatim, which was not really intended but is fine with me, and I think its explicit use of "Try to avoid" and "Considering using alternatives" is no weaker than using the word `recommend` explicitly, while both already imply the same, compare:
 - It is recommended not to cross the street
 - Try not to cross the street
 - Consider not crossing the street

The first statement feels stronger than the others, as 'recommending' claims authority, while 'trying' and 'considering' put the initiative with the reader. I'm not a real native English speaker myself so I won't try to be authoritative on the last nuances though  😉

Reply:
@curry684 I agree with your point. I still prefer `recommending` because I prefer to keep some level of authority but based on your argument I am also fine with `trying` and `considering`.

Reply:
Updated 👍